### PR TITLE
Prep v0.12.7 release

### PR DESCRIPTION
**Pull request by jedevc**
Changes since last release: https://github.com/dagger/dagger/compare/v0.12.6...main

### CODE REVIEW

**Comment by jedevc** [comment]
note: helm test is expected to fail since it's trying to pull the unreleased v0.12.7 tag. I need to fix this, but haven't had time since the last release.

File: .changes/v0.12.7.md
Comment (gerhard) [inline .changes/v0.12.7.md:6]: This reads as 2 fixed, when there's just one. Want to merge & maybe reword @jedevc?

File: helm/dagger/CHANGELOG.md
Comment (gerhard) [inline helm/dagger/CHANGELOG.md:12]: We should not have releases with no changes. The change worth mentioning here is that the Engine version now defaults to `v0.m.p`, same as we do for SDKs that may not have any changes.

**Review by gerhard — APPROVED** [review]
2 changes worth making before merging, but otherwise LGTM.

Approving so that this gets expedited 💨

File: .changes/v0.12.7.md
Comment (gerhard) [reply .changes/v0.12.7.md:6]: OK!

File: helm/dagger/CHANGELOG.md
Comment (jedevc) [reply helm/dagger/CHANGELOG.md:12]: Resolved, will update the release guide for next time.

**Pull request merged**
